crates/common/tedge_config/src/tedge_config_cli/tedge_config.rs

```diff
@@ -1047,6 +1048,29 @@ pub struct MqttAuthClientConfig {
     pub key_file: Utf8PathBuf,
 }
 
+impl TEdgeConfigReaderHttp {
+    pub fn client_tls_config(&self) -> anyhow::Result<Option<rustls::ClientConfig>> {
+        let client_cert_key = crate::all_or_nothing((
+            self.client.auth.cert_file.as_ref(),
+            self.client.auth.key_file.as_ref(),
+        ))
+        .map_err(|e| anyhow!("{e}"))?;
+
+        client_cert_key
+            .map(|(cert, key)| {
+                create_tls_config(
+                    self.ca_path
+                        .or_none()
+                        .map_or(DEFAULT_ROOT_CERT_PATH, |ca| ca.as_str()),
+                    cert,
+                    key,
```

Comment:
`create_tls_config` expects the key first and the cert second.

```suggestion
                    key,
                    cert,
```